ludwig/automl/auto_tune_config.py

```diff
@@ -0,0 +1,164 @@
+import copy
+from collections import OrderedDict
+
+import psutil
+import ray
+
+try:
+    import GPUtil
+except ImportError:
+    raise ImportError(
+        ' GPUtil is not installed. '
+        'In order to use this module please run '
+        'pip install GPUtil'
+    )
+
+from ludwig.api import LudwigModel
+from ludwig.automl.utils import get_available_resources
+from ludwig.data.preprocessing import preprocess_for_training
+from ludwig.features.feature_registries import update_config_with_metadata
+from ludwig.utils.defaults import merge_with_defaults
+from ludwig.constants import COMBINER, HYPEROPT, BATCH_SIZE, TRAINING, TYPE, PREPROCESSING, SPACE
+
+# maps variable search space that can be modified to minimum permissible value for the range
+RANKED_MODIFIABLE_PARAM_LIST = {
+    'tabnet': OrderedDict({
+        'training.batch_size': 32,
+        'combiner.size': 8,
+        'combiner.output_size': 8,
+    }),
+    'concat': OrderedDict({
+        'training.batch_size': 32,
+        'combiner.fc_size': 64,
+        'combiner.num_fc_layers': 1,
+
+    }),
+    'tabtransformer': OrderedDict({
+        'training.batch_size': 32,
+        'combiner.num_heads:': 4,
+        'combiner.output_size': 8,
+        'combiner.num_layers': 4,
+        'combiner.num_fc_layers': 1,
+    }),
+}
+
+
+BYTES_TO_MB = 1e6
+
+
+def get_trainingset_metadata(config, dataset):
+    (_, _, _, training_set_metadata) = preprocess_for_training(
+        config,
+        dataset=dataset,
+        preprocessing_params=config[PREPROCESSING])
+    return training_set_metadata
+
+
+def get_machine_memory():
+
+    if ray.is_initialized():  # using ray cluster
+        @ray.remote(num_gpus=1)
+        def get_remote_gpu():
+            gpus = GPUtil.getGPUs()
+            total_mem = gpus[0].memory_total
+            return total_mem * BYTES_TO_MB
```

Comment:
This calculation is a little bit confusing, maybe use `total_mem_mb * BYTES_PER_MB` so it's clear we're converting from MB to bytes.